feat(next): astro:routes:resolved (#12329)
withastro/astro: .changeset/giant-ravens-look.md

---
'astro': minor
---

Adds a new `astro:routes:resolved` hook to the Integration API, and deprecates `routes` passed to `astro:build:done`

When building an integration, you can now get access to routes inside the `astro:routes:resolved` hook:

```js
const integration = () => {
    return {
        name: 'my-integration',
        hooks: {
            'astro:routes:resolved': ({ routes }) => {
                console.log(routes)
            }
        }
    }
}
```

This hook runs before `astro:config:done`, and whenever a route changes in development.

> `routes` from `astro:build:done` is now deprecated. We recommend you use the new hook instead:

Reviewer: This goes to `next`, can't we do a clean cut?

Reviewer 2: Would break things on adapters and integrations. I think we can clean it before v5 directly but I think it would be better on a separate PR so it can be deprecated for a small period during the beta

Author: I'd be in favor of deprecating some properties in 5.0, and removing in 6.0. Deprecating in 5.0 beta and removing in stable feels too fast to me, especially since I think this PR will land closer to stable release

---
'astro': minor
---

Adds a new `astro:routes:resolved` hook to the Integration API, and deprecates most `routes` properties passed to `astro:build:done`

When building an integration, you can now get access to routes inside the `astro:routes:resolved` hook:

```js
const integration = () => {
    return {
        name: 'my-integration',
        hooks: {
            'astro:routes:resolved': ({ routes }) => {
                console.log(routes)
            }
        }
    }
}
```

This hook runs before `astro:config:done`, and whenever a route changes in development.

> Most `routes` properties from `astro:build:done` are now deprecated. We recommend you use the new hook instead:

Author: I think that's a good idea, if it's the only property that needs to be exposed. If we go down this path, I think it should be tackled in this PR

---
'astro': minor
---

> Adds a new `astro:routes:resolved` hook to the Integration API. Makes updates to the `astro:build:done` hook by deprecating `routes` and adding a new `assets` map

Reviewer 3: ```suggestion
Adds a new `astro:routes:resolved` hook to the Integration API. Also update the `astro:build:done` hook by deprecating `routes` and adding a new `assets` map.
```

When building an integration, you can now get access to routes inside the `astro:routes:resolved` hook:

```js
const integration = () => {
    return {
        name: 'my-integration',
        hooks: {
            'astro:routes:resolved': ({ routes }) => {
                console.log(routes)
            }
        }
    }
}
```

This hook runs before `astro:config:done`, and whenever a route changes in development.

> The `routes` array from `astro:build:done` are now deprecated. Any useful property is made available on `astro:routes:resolved`, except for `distURL`. Instead, you can use the newly exposed `assets` map::

Reviewer 3: ```suggestion
The `routes` array from `astro:build:done` is now deprecated, and exposed properties are now available on `astro:routes:resolved`, except for `distURL`. For this, you can use the newly exposed `assets` map:
```
